tests/ttnn/unit_tests/tensor/test_tensor_creation.py

```diff
@@ -120,3 +120,143 @@ def test_tensor_creation_api_parity(shape, tt_dtype, layout, device):
+    "tt_dtype",
+    [
+        ttnn.uint8,
+        ttnn.uint16,
+        ttnn.uint32,
+        ttnn.int32,
+        ttnn.float32,
+        ttnn.bfloat16,
+        ttnn.bfloat8_b,
+        ttnn.bfloat4_b,
+    ],
+)
+@pytest.mark.parametrize(
+    "shape, memory_config",
+    [
+        (
+            (1, 2, 3, 4),
+            ttnn.MemoryConfig(
+                ttnn.TensorMemoryLayout.INTERLEAVED,
+                ttnn.BufferType.DRAM,
+            ),
+        ),
+        (
+            (1, 48, 56, 32),
+            ttnn.MemoryConfig(
+                ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
+                ttnn.BufferType.L1,
+                ttnn.ShardSpec(
+                    ttnn.num_cores_to_corerangeset(56, grid_size, True),
+                    [48, 32],
+                    ttnn.ShardOrientation.ROW_MAJOR,
+                    False,
+                    ttnn.ShardMode.LOGICAL,
+                ),
+            ),
+        ),
+        (
+            (1, 2, 10, 5),
+            ttnn.MemoryConfig(
+                ttnn.TensorMemoryLayout.WIDTH_SHARDED,
+                ttnn.BufferType.L1,
+                ttnn.ShardSpec(
+                    ttnn.num_cores_to_corerangeset(3, grid_size, True),
+                    [20, 2],
+                    ttnn.ShardOrientation.ROW_MAJOR,
+                    False,
+                    ttnn.ShardMode.LOGICAL,
+                ),
+            ),
+        ),
+        (
+            (2, 3, 64, 96),
+            ttnn.MemoryConfig(
+                ttnn.TensorMemoryLayout.BLOCK_SHARDED,
+                ttnn.BufferType.L1,
+                ttnn.ShardSpec(
+                    ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(1, 5))}),
+                    [64, 64],
+                    ttnn.ShardOrientation.ROW_MAJOR,
+                    False,
+                    ttnn.ShardMode.LOGICAL,
+                ),
+            ),
+        ),
+        (
+            (1, 8, 36, 32),
+            ttnn.MemoryConfig(
+                ttnn.TensorMemoryLayout.BLOCK_SHARDED,
+                ttnn.BufferType.L1,
+                ttnn.ShardSpec(
+                    ttnn.CoreRangeSet({ttnn.CoreRange(ttnn.CoreCoord(0, 0), ttnn.CoreCoord(3, 5))}),
+                    [48, 10],
+                    [64, 64],  # NOTE: This value is compatible with all PageConfigs in this sweep
+                    ttnn.ShardOrientation.ROW_MAJOR,
+                    False,
+                ),
+            ),
+        ),
+    ],
+    ids=[
+        "interleaved",
+        "height_sharded",
+        "width_sharded",
+        "block_sharded",
+        "block_sharded_with_custom_physical_shard_shape",
+    ],
+)
+def test_tensor_creation_with_memory_config(shape, memory_config, tt_dtype, layout, tile, device):
+    torch.manual_seed(0)
+
+    if tt_dtype in (ttnn.bfloat8_b, ttnn.bfloat4_b) and layout == ttnn.ROW_MAJOR_LAYOUT:
+        pytest.skip("{} is only valid for ttnn.TILE_LAYOUT!".format(tt_dtype))
```

Comment:
I wonder, does the code gracefully handle these cases, throwing a nice and clear error message?

Reply:
- `ttnn.Tensor(...)` and `tensor.to()` will throw warning and still work (as in, create as TILE layout). This was to support legacy code that attempted to create `ttnn.bfloat8_b` or `ttnn.bfloat4_b` in ROW_MAJOR layout. This should be cleaned up and turned into fatal errors instead.
- `ttnn.from_torch()` and `ttnn.to_torch()` will throw error right away (as it should).

Reply:
Brian, how much work is it to convert the legacy code to use the correct layout? Looking at the override [here](https://github.com/tenstorrent/tt-metal/blob/05fee29670603f4dc8e5cd847c0609dc57e407ef/ttnn/cpp/pybind11/pytensor.cpp#L412-L423).

Reply:
Idk, it's in lots of places